R/toolbar.R

```diff
@@ -136,10 +150,216 @@ toolbar_input_button <- function(
     tooltip <- NULL
   }
   if (!is.null(tooltip)) {
-    # Default placement is "bottom" for the toolbar case because otherwise the
-    # tooltip ends up covering the neighboring buttons in the header/footer.
+    # Default placement is "bottom" for the toolbar case because
+    # otherwise the tooltip ends up covering the neighboring buttons in
+    # the header/footer.
     button <- tooltip(button, tooltip, placement = "bottom")
   }
 
   button
 }
+
+#' Toolbar Input Select
+#'
+#' @description
+#' Create a select list input control that can be used to choose a single
+#' item from a list of values, suitable for use within a [toolbar()].
+#'
+#' @examplesIf rlang::is_interactive()
+#' toolbar(
+#'   align = "right",
+#'   toolbar_input_select(
+#'     id = "select",
+#'     label = "Choose option",
+#'     choices = c("Option 1", "Option 2", "Option 3"),
+#'     selected = "Option 2"
+#'   )
+#' )
+#'
+#' # With tooltip
+#' toolbar(
+#'   align = "right",
+#'   toolbar_input_select(
+#'     id = "select",
+#'     label = "Choose option",
+#'     choices = c("Option 1", "Option 2", "Option 3"),
+#'     tooltip = "Select your preferred option from the list"
+#'   )
+#' )
+#'
+#' # With icon and tooltip
+#' toolbar(
+#'   align = "right",
+#'   toolbar_input_select(
+#'     id = "select",
+#'     label = "Choose option",
+#'     choices = c("Option 1", "Option 2", "Option 3"),
+#'     icon = shiny::icon("filter"),
+#'     tooltip = "Filter the data"
+#'   )
+#' )
+#'
+#' @param id The input ID.
+#' @param label The label for the select input. A hidden label is created with
+#'   this text and referenced via `aria-labelledby` for accessibility. Must be
+#'   a non-empty string.
+#' @param choices List of values to select from. If elements of the list
+#'   are named, then that name — rather than the value — is displayed to
+#'   the user. It's also possible to group related inputs by providing a
+#'   named list whose elements are (either named or unnamed) lists,
+#'   vectors, or factors. In this case, the outermost names will be used
+#'   as the group labels (leveraging the ⁠<optgroup>⁠ HTML tag) for the
+#'   elements in the respective sublist.
+#' @param selected The initially selected value. If not provided, the first
+#'   choice will be selected by default.
+#' @param tooltip Tooltip text to display when hovering over the select
+#'   input. Can be:
+#'   * `NULL` (default) - no tooltip is shown
+#'   * A character string - wraps the entire input with a tooltip
```

Comment:
Since there are only two options, it'd be better to just use prose here. Some editors only show the first paragraph of text in hover tips and might miss the list, so I try to only use them when it's structurally important or easier to read.

Reply:
Also this behavior should be inline with what we've established for `toolbar_input_button(tooltip=)`